backend/app/core/security.py

```
@@ -176,4 +176,26 @@ def get_username_from_request(request) -> str:
         token_data = verify_token(token)
         return token_data.get("username", "anonymous")
     except Exception:
-        return "invalid_token"
\ No newline at end of file
+        return "invalid_token"
+
+def get_admin_user(current_user: User = Depends(get_current_user)) -> User:
```

Ruff (0.14.2)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)